src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/CreateSdkSymbolsLayout.cs

```diff
@@ -0,0 +1,165 @@
+                Log.LogError(message);
+            }
+            else
+            {
+                Log.LogWarning(message);
+            }
+        }
+
+        private IList<string> GenerateSymbolsLayout(string sdkRoot, string destinationRoot)
+        {
+            List<string> filesWithoutPDBs = new List<string>();
+
+            if (Directory.Exists(destinationRoot))
+            {
+                Directory.Delete(destinationRoot, true);
+            }
+
+            foreach (string file in Directory.GetFiles(sdkRoot, "*", SearchOption.AllDirectories))
+            {
+                if (file.EndsWith(".dll", StringComparison.InvariantCultureIgnoreCase) &&
+                    !file.EndsWith(".resources.dll", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    string guid = string.Empty;
+                    using var pdbStream = File.OpenRead(file);
+                    using var peReader = new PEReader(pdbStream);
+                    try
+                    {
+                        // Check if pdb is embedded
+                        if (peReader.ReadDebugDirectory().Any(entry => entry.Type == DebugDirectoryEntryType.EmbeddedPortablePdb))
+                        {
+                            continue;
+                        }
+
+                        var debugDirectory = peReader.ReadDebugDirectory().First(entry => entry.Type == DebugDirectoryEntryType.CodeView);
+                        var codeViewData = peReader.ReadCodeViewDebugDirectoryData(debugDirectory);
+                        guid = $"{codeViewData.Guid.ToString("N").Replace("-", string.Empty)}";
+                    }
+                    catch (Exception)
+                    {
+                        // Ignore binaries without debug info
+                        continue;
+                    }
+
+                    if (guid != string.Empty)
+                    {
+                        if (!AllPdbGuids.ContainsKey(guid))
+                        {
+                            filesWithoutPDBs.Add(file.Substring(sdkRoot.Length + 1));
+                        }
+                        else
+                        {
+                            // Copy matching pdb to symbols path, preserving sdk binary's hierarchy
+                            string sourcePath = (string)AllPdbGuids[guid]!;
+                            string destinationPath =
+                                file.Replace(sdkRoot, destinationRoot)
+                                    .Replace(Path.GetFileName(file), Path.GetFileName(sourcePath));
+
+                            Directory.CreateDirectory(Path.GetDirectoryName(destinationPath)!);
+                            File.Copy(sourcePath, destinationPath, true);
+                        }
+                    }
+                }
+            }
+
+            return filesWithoutPDBs;
+        }
+
+        public void IndexAllSymbols(string path)
+        {
+            AllPdbGuids = new Hashtable();
```

Comment:
From a design perspective, what is the reason for tracking AllPdbGuids as an instance versus returning this from this method?

Reply:
Fixed in https://github.com/dotnet/installer/pull/17454/commits/b9f64fd009e8a3be9ad733950302f7cf7d963116